tasks/main.yml

```diff
@@ -1,3 +1,112 @@
 ---
+- name: Ensure database is created
+  become: true
+  become_user: postgres
+  postgresql_db:
+    name: "{{ mattermost_db_name }}"
 
-...
+- name: Ensure user has access to the database
+  become: yes
+  become_user: postgres
+  postgresql_user:
+    db: "{{ mattermost_db_name }}"
+    name: "{{ mattermost_db_user }}"
+    password: "{{ mattermost_db_password }}"
+    priv: ALL
+
+- name: Ensure user does not have unnecessary privilege
+  become: yes
+  become_user: postgres
+  postgresql_user: 
+    name: "{{ mattermost_db_user }}"
+    role_attr_flags: NOSUPERUSER,NOCREATEDB
+
+- name: 'Download binary from https://releases.mattermost.com'
+  get_url: 
+    url: 'https://releases.mattermost.com/{{ mattermost_version }}/mattermost-team-{{ mattermost_version }}-linux-amd64.tar.gz'
+    dest: /tmp/
```

Comment:
use the `global_cache_dir` variable we use across our roles. Look at the vault or zfs-snap-manager role to download the archive on the executing host. Hereby, the target machine does not need internet access.

Reply:
should be fixed